### PR TITLE
MPT-22563 AWS: migrate Teams notifications to Workflow webhook Adaptive Cards

### PR DESCRIPTION
**Pull request by ruben-sebrango**
🤖 AI-generated PR — Please review carefully.

## What

Migrate MS Teams notifications from the deprecated incoming webhook connector to Power Automate Workflow webhooks using Adaptive Cards.

- Replace `pymsteams` connector card calls with `requests.post` sending Adaptive Card JSON directly to the Workflow webhook URL.
- Remove the `pymsteams` dependency from `pyproject.toml` and `uv.lock`.
- Introduce module-level style constants (`_WARNING`, `_SUCCESS`, `_ATTENTION`) and a `_build_card` method that constructs the Adaptive Card payload.
- Add `# noqa: WPS229` on the two-statement try block (post + raise_for_status), which is the canonical HTTP request pattern.
- Update `docs/external/ms-teams.md` and `docs/external-integrations.md` to reflect the new Workflow webhook integration.
- Rewrite `tests/swo/notifications/test_teams.py` to mock `requests.post` and assert on the Adaptive Card payload instead of the old `pymsteams` API.
- Add a missing test for the duplicate-agreement validation error (`AWS004`) in `tests/flows/validation/test_base.py`.

## Why

Microsoft deprecated the Office 365 Connector (incoming webhooks) used by `pymsteams`. The replacement is a Power Automate Workflow webhook that expects Adaptive Card payloads. Without this change, Teams notifications stop working when the connector is decommissioned.

## Testing

- `make check-all` passes: 1084 tests, 0 failures, 99% coverage.
- `teams.py` reaches 98% coverage with the new test suite.

Closes [MPT-22563](https://softwareone.atlassian.net/browse/MPT-22563)

- Replaced deprecated `pymsteams` incoming webhook connector-card flow with Microsoft Power Automate **Workflow webhooks** by sending Adaptive Card JSON via direct `requests.post`.
- Removed `pymsteams` from `pyproject.toml` (and the related mypy override).
- Refactored Teams notification implementation to use module-level style constants (`Style`), a `_build_card(...)` helper for Adaptive Card construction, and centralized POST/`raise_for_status()` handling with request-timeout and updated logging.
- Updated MS Teams and integrations documentation to reflect Workflow webhook (no auth) setup and the Adaptive Card/style format.
- Updated Teams notification tests to mock `requests.post` and assert the generated Adaptive Card payload and error handling behavior.
- Added missing validation test coverage for the duplicate-agreement error `AWS004`.

[MPT-22563]: https://softwareone.atlassian.net/browse/MPT-22563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swo_aws_extension/swo/notifications/teams.py [hunk 1]
```diff
@@ -2,11 +2,16 @@
 import logging
 from dataclasses import dataclass
 
-import pymsteams
+import requests
 from django.conf import settings
 
 logger = logging.getLogger(__name__)
 
+_WARNING = ("warning", "Warning")
```
Comment (jentyk) [inline swo_aws_extension/swo/notifications/teams.py]: Maybe it is worth making it a Style `Enum`? Just to make it clear what this is for?

**Review by jentyk — APPROVED** [review]
